Improve Epoxy upgrade doc (#1841)
stackhpc/stackhpc-kayobe-config: doc/source/operations/upgrading-openstack.rst

----------------------

Ubuntu Jammy support has been removed from the 2025.1 release onwards. Hosts
must be migrated to Ubuntu 24.04 before upgrading OpenStack services.
You can find the upgrade procedure from :ref:`upgrading-to-ubuntu-noble`
documentation.

> RabbitMQ 4.1
> ------------
> 
> .. warning::
> 
>    StackHPC Kayobe Config sets RabbitMQ 4.1 as the default for the Epoxy release.
>    Existing transient queues must be migrated to durable queues with Queue Manager
>    before upgrading to RabbitMQ 4.1.

Reviewer: Apologies, I've actually changed my mind.

I think if we change the heading, and clearly state the order of operations, it will make it much clearer for the reader.

```suggestion

RabbitMQ Prerequisites
----------------------

.. warning::

   StackHPC Kayobe Config sets RabbitMQ 4.1 as the default for the Epoxy release.
   Existing transient queues must be migrated to durable queues with Queue Manager
   before upgrading to RabbitMQ 4.1.
   
   This means that queue migration and the RabbitMQ 4.1 upgrade must be completed
   before upgrading to Epoxy.
```

Author: Do you want the second paragraph to be inside of warning block or not?

Reviewer: Inside I think

Author: Then we need to remove line 301 right?

Reviewer: It actually renders fine as is:
<img width="1125" height="365" alt="image" src="https://github.com/user-attachments/assets/d8aed5be-c60a-4a5d-a15c-819a3691404d" />
I'd prefer to keep it as two separate paragraphs if possible

----------------------

Ubuntu Jammy support has been removed from the 2025.1 release onwards. Hosts
must be migrated to Ubuntu 24.04 before upgrading OpenStack services.
You can find the upgrade procedure from :ref:`upgrading-to-ubuntu-noble`
documentation.

RabbitMQ 4.1
------------

.. warning::

>    RabbitMQ queue migration and upgrade must be done in Caracal environment.

Reviewer: ```suggestion
   The RabbitMQ queue migration and upgrade must be completed before upgrading the environment to Epoxy.
```

StackHPC Kayobe Config sets RabbitMQ 4.1 as the default for the Epoxy release.
Existing transient queues must be migrated to durable queues with Queue Manager

> before upgrading to RabbitMQ 4.1.

Reviewer: This says the same thing as the warning, so can be removed. If you want to keep some of the detail, I'd add it to the warning instead

Author: Maybe this thing should be in warning instead of the warning above it as this section being part of Prerequisites implies that this needs to be done before upgrade.

Reviewer: Sounds sensible. I'll leave it up to you how you resolve it, I just think that at the moment, both sections say the same thing

Queue Migration
~~~~~~~~~~~~~~~

.. warning::

   This migration will stop all services using RabbitMQ and cause an extended
   API outage while queues are migrated. It should only be performed in a
   pre-agreed maintenance window.

   If you are using Azimuth or the ClusterAPI driver for Magnum, you should
   make sure to pause reconciliation of all clusters before the API outage
   window. See the `Azimuth docs
   <https://azimuth-cloud.github.io/azimuth-config/operations/maintenance/>`__
   for instructions.

Set the following variables in your kolla globals file (i.e.

> $KAYOBE_CONFIG_PATH/kolla/globals.yml or $KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/kolla/globals.yml):

Reviewer: ```suggestion
``$KAYOBE_CONFIG_PATH/kolla/globals.yml`` or ``$KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/kolla/globals.yml``):
```

.. code-block:: yaml

   om_enable_queue_manager: true
   om_enable_rabbitmq_quorum_queues: true
   om_enable_rabbitmq_transient_quorum_queue: true
   om_enable_rabbitmq_stream_fanout: true

Then execute the migration script:

.. code-block:: bash

   $KAYOBE_CONFIG_PATH/../../tools/rabbitmq-queue-migration.sh

RabbitMQ Upgrade
~~~~~~~~~~~~~~~~

> After queue migration is finished, upgrade RabbitMQ to 4.1.

Reviewer: ```suggestion
After the queue migration is finished, upgrade RabbitMQ to 4.1.
```

1. Sync and publish latest Kolla container images to ensure local pulp has RabbitMQ 4.1 image.

>    (This can be skipped if local pulp is not used.)

Reviewer: Does anywhere not have a local pulp instance?

Author: Probably not but I added that for someone who doesn't for whatever reason
